### ISSUE TITLE
Unable to train the keypointrcnn_resnet50_fpn model

### ISSUE THREAD

**Issue by d1nz-g33k** [pytorch/vision#3801]
### I'm trying to train the model for keypoints detection, & bounding boxes, using the script in the references section, while training the loss_keypoint is always 0.0000. Using those weights for prediction is giving no predictions at all.

I'm running this on a Windows 2016 server (EC2 instance on AWS), with a single GPU (Instance Type: p2.xlarge)

My Dataset is in COCO format, but I'm using only 14 keypoints per person, so I had defined the model in the train.py file as below:
```
model = torchvision.models.detection.keypointrcnn_resnet50_fpn(
        pretrained=False, progress=True, num_classes=1, num_keypoints=14, 
        pretrained_backbone=True, trainable_backbone_layers=None)
```

& I've made appropriate changes in coco_utils.py for Keypoint flip.

**Training**
Command:
```
python train.py --dataset coco_kp2 --model keypointrcnn_resnet50_fpn --epochs 1 --lr-steps 36 43 
--aspect-ratio-group-factor 3
```
Output:
```
Not using distributed mode
Namespace(aspect_ratio_group_factor=3, batch_size=2, data_augmentation='hflip', data_path='/datasets01/COCO/022719/', dataset='coco_kp2', device='cuda', dist_url='env://', distributed=False, epochs=1, lr=0.02, lr_gamma=0.1, lr_step_size=8, lr_steps=[36, 43], model='keypointrcnn_resnet50_fpn', momentum=0.9, output_dir='.', pretrained=False, print_freq=20, resume='', rpn_score_thresh=None, start_epoch=0, test_only=False, trainable_backbone_layers=None, weight_decay=0.0001, workers=4, world_size=1)
Loading data
loading annotations into memory...
Done (t=0.02s)
creating index...
index created!
loading annotations into memory...
Done (t=0.02s)
creating index...
index created!
Creating data loaders
Using [0, 0.5, 0.6299605249474366, 0.7937005259840997, 1.0, 1.2599210498948732, 1.5874010519681994, 2.0, inf] as bins for aspect ratio quantization
Count of instances per bin: [180]
Creating model
Start training
Epoch: [0]  [ 0/90]  eta: 0:11:55  lr: 0.000244  loss: 0.7178 (0.7178)  loss_classifier: 0.0000 (0.0000)  loss_box_reg: 0.0000 (0.0000)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 0.6962 (0.6962)  loss_rpn_box_reg: 0.0216 (0.0216)  time: 7.9505  data: 5.2040  max mem: 2618
Epoch: [0]  [20/90]  eta: 0:02:04  lr: 0.004734  loss: 0.6764 (0.6253)  loss_classifier: 0.0000 (0.0000)  loss_box_reg: 0.0000 (0.0000)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 0.6526 (0.6053)  loss_rpn_box_reg: 0.0186 (0.0200)  time: 1.4630  data: 0.0062  max mem: 2951
Epoch: [0]  [40/90]  eta: 0:01:22  lr: 0.009224  loss: 0.0664 (0.3587)  loss_classifier: 0.0000 (0.0000)  loss_box_reg: 0.0000 (0.0000)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 0.0488 (0.3400)  loss_rpn_box_reg: 0.0147 (0.0186)  time: 1.5132  data: 0.0061  max mem: 2951
Epoch: [0]  [60/90]  eta: 0:00:47  lr: 0.013714  loss: 0.0196 (0.2480)  loss_classifier: 0.0000 (0.0000)  loss_box_reg: 0.0000 (0.0000)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 0.0072 (0.2316)  loss_rpn_box_reg: 0.0118 (0.0164)  time: 1.4801  data: 0.0065  max mem: 2951
Epoch: [0]  [80/90]  eta: 0:00:15  lr: 0.018204  loss: 0.0192 (0.1919)  loss_classifier: 0.0000 (0.0000)  loss_box_reg: 0.0000 (0.0000)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 0.0067 (0.1761)  loss_rpn_box_reg: 0.0121 (0.0158)  time: 1.4868  data: 0.0062  max mem: 2951
Epoch: [0]  [89/90]  eta: 0:00:01  lr: 0.020000  loss: 0.0182 (0.1745)  loss_classifier: 0.0000 (0.0000)  loss_box_reg: 0.0000 (0.0000)  loss_keypoint: 0.0000 (0.0000)  loss_objectness: 0.0067 (0.1591)  loss_rpn_box_reg: 0.0107 (0.0153)  time: 1.4933  data: 0.0053  max mem: 2951
Epoch: [0] Total time: 0:02:20 (1.5584 s / it)
Test:  [ 0/90]  eta: 0:08:32  model_time: 0.4447 (0.4447)  evaluator_time: 0.0010 (0.0010)  time: 5.6930  data: 5.2317  max mem: 2951
Test:  [89/90]  eta: 0:00:00  model_time: 0.3594 (0.3613)  evaluator_time: 0.0010 (0.0011)  time: 0.3689  data: 0.0033  max mem: 2951
Test: Total time: 0:00:38 (0.4315 s / it)
Averaged stats: model_time: 0.3594 (0.3613)  evaluator_time: 0.0010 (0.0011)
Accumulating evaluation results...
DONE (t=0.02s).
Accumulating evaluation results...
DONE (t=0.00s).
IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
IoU metric: keypoints
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.000
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.000
Training time 0:03:02
```

I'm using the .pth file generated during the trining in the **keypoint_rcnn_videos.py** 
Command:
```python keypoint_rcnn_videos.py --input ../input/video3.mp4```
Output:
```
cuda
[{'boxes': tensor([], device='cuda:0', size=(0, 4)), 'labels': tensor([], device='cuda:0', dtype=torch.int64), 'scores': tensor([], device='cuda:0'), 'keypoints': tensor([], device='cuda:0', size=(0, 14, 3)), 'keypoints_scores': tensor([], device='cuda:0', size=(0, 14))}]
[{'boxes': tensor([], device='cuda:0', size=(0, 4)), 'labels': tensor([], device='cuda:0', dtype=torch.int64), 'scores': tensor([], device='cuda:0'), 'keypoints': tensor([], device='cuda:0', size=(0, 14, 3)), 'keypoints_scores': tensor([], device='cuda:0', size=(0, 14))}]
[{'boxes': tensor([], device='cuda:0', size=(0, 4)), 'labels': tensor([], device='cuda:0', dtype=torch.int64), 'scores': tensor([], device='cuda:0'), 'keypoints': tensor([], device='cuda:0', size=(0, 14, 3)), 'keypoints_scores': tensor([], device='cuda:0', size=(0, 14))}]
[{'boxes': tensor([], device='cuda:0', size=(0, 4)), 'labels': tensor([], device='cuda:0', dtype=torch.int64), 'scores': tensor([], device='cuda:0'), 'keypoints': tensor([], device='cuda:0', size=(0, 14, 3)), 'keypoints_scores': tensor([], device='cuda:0', size=(0, 14))}]
```
I'm just printing the device, to check if it's using the GPU - Hence the **_cuda_** in the top
After that I'm printing the output from model for each frame. It repeats the same for the whole video.

I've also added `if masks is not None:` condition, wherever required in **coco_utils.py**, since my annotations don't contain segmentations.

I've tried training the model for higher epochs (till 13), yet no change. Also tried changing the classes to 2 (1 for background), yet the same result.
Please point out what I'm doing wrong. Thanks a lot!

PS: This is my first time posting a question, so please feel free to correct me. Thanks!
